File: android/src/withMaps/java/org/ligi/passandroid/maps/PassbookMapsFacade.java
package org.ligi.passandroid.maps;


import android.content.Context;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentTransaction;

import com.google.android.gms.common.ConnectionResult;
import com.google.android.gms.common.GooglePlayServicesUtil;

import org.ligi.axt.AXT;
import org.ligi.passandroid.App;
import org.ligi.passandroid.FullscreenMapActivity;
import org.ligi.passandroid.LocationsMapFragment;
import org.ligi.passandroid.R;
import org.ligi.passandroid.model.Pass;

public class PassbookMapsFacade {

    public static boolean init(FragmentActivity context) {
        final boolean isGooglePlayServicesAvailable = GooglePlayServicesUtil.isGooglePlayServicesAvailable(context) == ConnectionResult.SUCCESS;

        if (!isGooglePlayServicesAvailable) {
            return false;
        }

        FragmentTransaction ft = context.getSupportFragmentManager().beginTransaction();
        LocationsMapFragment locationsMapFragment = new LocationsMapFragment();
        locationsMapFragment.click_to_fullscreen = true;
        ft.replace(R.id.map_container, locationsMapFragment);
        ft.commit();

        return true;
    }

    public static void startFullscreenMap(Context context, Pass pass) {
        App.getPassStore().setCurrentPass(pass);
        AXT.at(context).startCommonIntent().activityFromClass(FullscreenMapActivity.class);
    }
}

File: android/src/withMaps/java/org/ligi/passandroid/maps/PassbookMapsFacade.kt
package org.ligi.passandroid.maps

import android.content.Context
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.commit
import androidx.fragment.app.replace
import com.google.android.gms.common.ConnectionResult
import com.google.android.gms.common.GoogleApiAvailability
import org.ligi.kaxt.startActivityFromClass
import org.ligi.passandroid.FullscreenMapActivity
import org.ligi.passandroid.LocationsMapFragment
import org.ligi.passandroid.R

object PassbookMapsFacade {

    fun init(context: FragmentActivity): Boolean {
        val gapi = GoogleApiAvailability.getInstance()
        val isGooglePlayServicesAvailable = gapi.isGooglePlayServicesAvailable(context) == ConnectionResult.SUCCESS

        if (!isGooglePlayServicesAvailable) {
            return false
        }

        val locationsMapFragment = LocationsMapFragment()
        locationsMapFragment.clickToFullscreen = true
        context.supportFragmentManager.commit { replace<LocationsMapFragment>(R.id.map_container) }

        return true
    }

    fun startFullscreenMap(context: Context) {
        context.startActivityFromClass(FullscreenMapActivity::class.java)
    }
}